centreon/src/Core/Notification/Application/UseCase/FindNotifications/FindNotifications.php

```diff
@@ -0,0 +1,280 @@
+                new ForbiddenResponse(NotificationException::listNotAllowed())
+            );
+
+            return;
+        }
+
+        try {
+            $this->info('Search for Notifications');
+            $notifications = $this->notificationRepository->findAll($this->requestParameters);
+            if (empty($notifications)) {
+                $presenter->presentResponse(new FindNotificationsResponse());
+
+                return;
+            }
+
+            $notificationsIds = [];
+            foreach ($notifications as $notification) {
+                $notificationsIds[] = $notification->getId();
+            }
+            $this->info(
+                'Retrieving Notification channels for notifications',
+                ['notifications' => implode(', ', $notificationsIds)]
+            );
+            $notificationChannelByNotifications = $this->notificationRepository
+                ->findNotificationChannelsByNotificationIds(
+                    $notificationsIds
+                );
+            $notificationCounts = $this->getCountByNotifications($notificationsIds);
+            $presenter->presentResponse(
+                $this->createResponse($notifications, $notificationCounts, $notificationChannelByNotifications)
+            );
+        } catch (RequestParametersTranslatorException $ex) {
+            $presenter->presentResponse(new ErrorResponse($ex->getMessage()));
+            $this->error($ex->getMessage(), ['trace' => $ex->getTraceAsString()]);
+        } catch (NotificationException $ex) {
+            $this->error('An error occurred while retrieving the notifications listing', ['trace' => (string) $ex]);
+            $presenter->presentResponse(new ErrorResponse($ex));
+        } catch (\Throwable $ex) {
+            $this->error('An error occurred while retrieving the notifications listing', ['trace' => (string) $ex]);
+            $presenter->presentResponse(
+                new ErrorResponse(_('An error occurred while retrieving the notifications listing'))
+            );
+        }
+    }
+
+    /**
+     * Retrieve the count of users, hostgroup resources, servicegroup resources for the listed notifications.
+     *
+     * @param non-empty-array<int> $notificationsIds
+     *
+     * @return NotificationCounts
+     */
+    private function getCountByNotifications(array $notificationsIds): NotificationCounts
+    {
+        $this->info('Retrieving user counts for notifications', ['notification' => implode(', ', $notificationsIds)]);
+        $notificationsUsersCount = $this->notificationRepository->findUsersCountByNotificationIds(
+            $notificationsIds
+        );
+        if (($notificationIdWithEmptyUser = array_search(0, $notificationsUsersCount, true)) !== false) {
+            $this->error('No users found for a notification', ['notification' => $notificationIdWithEmptyUser]);
+
+            throw NotificationException::invalidUsers($notificationIdWithEmptyUser);
+        }
```

Comment:
This should be discuss with PM because we cannot create a notification without users, but if we delete all the contacts that are linked to a notification, the notification will not have users